lib/auth/v2/getCanonicalizedAmzHeaders.js

```diff
@@ -0,0 +1,44 @@
+'use strict'; // eslint-disable-line strict
+
+function getCanonicalizedAmzHeaders(headers) {
+    /*
+    Iterate through headers and pull any headers that are x-amz headers.
+    Need to include 'x-amz-date' here even though AWS docs
+    ambiguous on this.
+    */
+    const amzHeaders = Object.keys(headers)
+        .filter(val => val.substr(0, 6) === 'x-amz-')
```

Comment:
These lines are not affected by this PR, so I'd rather preserve the original code and make the minimal changes possible in order to avoid possible side effects. Does that sound good? ;)